src/Raven.Server/Commercial/LicenseManager.cs

```diff
@@ -121,40 +161,138 @@ public void Initialize(StorageEnvironment environment, TransactionContextPool co
             }
         }
 
-        public LicenseStatus GetLicenseStatus()
+        public async Task ReloadLicenseLimits()
         {
-            var lastLicenseIndex = _serverStore.LastLicenseIndex;
-            if (lastLicenseIndex > _lastSavedLicenseIndex)
+            try
             {
-                _lastSavedLicenseIndex = lastLicenseIndex;
-                ReloadLicense();
+                var licenseLimits = _serverStore.LoadLicenseLimits();
+                if (licenseLimits?.CoresByNode != null && 
+                    licenseLimits.CoresByNode.TryGetValue(_serverStore.NodeTag, out var cores))
+                {
+                    cores = Math.Min(cores, _licenseStatus.MaxCores);
+
+                    var process = Process.GetCurrentProcess();
+                    SetAffinity(process, cores);
+
+                    var ratio = (int)Math.Ceiling(_licenseStatus.MaxRamInGb / (double)_licenseStatus.MaxCores);
```

Comment:
`MaxRamInGB` isn't matched to `MaxCores`, I think.
That is the total GB on the cluster, but `MaxCores` was previously used to limit a single node